tests/lightning_qubit/test_state_vector_class.py

```diff
@@ -0,0 +1,470 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the serialization helper functions.
+"""
+
+import pytest
+from conftest import LightningDevice  # tested device
+
+import math
+import numpy as np
+import pennylane as qml
+from pennylane.wires import Wires
+from pennylane.tape import QuantumScript
+
+from pennylane_lightning.lightning_qubit._state_vector import LightningStateVector
+from pennylane_lightning.lightning_qubit import LightningQubit
+
+if not LightningQubit._CPP_BINARY_AVAILABLE:
+    pytest.skip("No binary module found. Skipping.", allow_module_level=True)
+
+if LightningDevice != LightningQubit:
+    pytest.skip("Exclusive tests for lightning.qubit. Skipping.", allow_module_level=True)
+
+
+@pytest.mark.parametrize("num_wires", [3, 10])
```

Comment:
To specify what to do with `0` and test the frequent cases `1` and `2`. 
```suggestion
@pytest.mark.parametrize("num_wires", range(4))
```

Reply:
Thank you for that!